Mathlib/Topology/LocalAtTarget.lean

```diff
@@ -65,15 +65,15 @@ alias ClosedEmbedding.restrictPreimage := Set.restrictPreimage_closedEmbedding
 
 theorem Set.restrictPreimage_isClosedMap (s : Set β) (H : IsClosedMap f) :
     IsClosedMap (s.restrictPreimage f) := by
-  rintro t ⟨u, hu, e⟩
-  refine' ⟨⟨_, (H _ (IsOpen.isClosed_compl hu)).1, _⟩⟩
-  rw [← (congr_arg HasCompl.compl e).trans (compl_compl t)]
-  simp only [Set.preimage_compl, compl_inj_iff]
-  ext ⟨x, hx⟩
-  suffices (∃ y, y ∉ u ∧ f y = x) ↔ ∃ y, y ∉ u ∧ f y ∈ s ∧ f y = x by
-    simpa [Set.restrictPreimage, ← Subtype.coe_inj]
-  exact ⟨fun ⟨a, b, c⟩ => ⟨a, b, c.symm ▸ hx, c⟩, fun ⟨a, b, _, c⟩ => ⟨a, b, c⟩⟩
-#align set.restrict_preimage_is_closed_map Set.restrictPreimage_isClosedMap
+  intro t ⟨u, hu, e⟩
+  refine' ⟨(f '' uᶜ)ᶜ, isOpen_compl_iff.2 (H uᶜ (IsOpen.isClosed_compl hu)), _⟩
+  simp [e, ← image_restrictPreimage]
+
+theorem Set.restrictPreimage_isOpenMap (s : Set β) (H : IsOpenMap f) :
+    IsOpenMap (s.restrictPreimage f) := by
+  intro t ⟨u, hu, e⟩
+  refine' ⟨f '' u, H u hu, _⟩
```

Comment:
You can condense this proof down to one line:
- use the trick above to replace the last two lines by `exact ⟨f '' u, H u hu, by rw [← e, image_restrictPreimage]⟩`
- now, you can switch to providing a term proof instead: instead of by, intro and exact you'd write just
`fun t ⟨u, hu, e⟩ ↦ ⟨f '' u, H u hu, by rw [← e, image_restrictPreimage]⟩` (with no `by` before).

(This is a bit of a detour/general comment. You don't have to memorize this; feel free to read on if you find this interesting.)
In general, it's a matter of style if condensing proofs this way is *desirable*. (If you write the shortest code in the world, but nobody can read it any more, that's not helpful for a library like mathlib which changes often.)
In this case, I'd say it is, as there is not much happening mathematically. In other cases, sometimes it still happens to me that I write code one way and a reviewer suggests a different style: developing "good taste" is always subjective and takes time.

Reply:
I agree that this proof and the previous one don't need to be three lines long, so I condensed them as you suggested (actually, in this case, I don't even find that shortening these proofs makes them less readable).